core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java

```diff
@@ -156,8 +168,24 @@ public ImageFromDockerfile withBuildArgs(final Map<String, String> args) {
         return this;
     }
 
+    /**
+     * Sets the Dockerfile to be used for this image. 
+     * It is recommended to use {@link #withDockerfile} instead because it honors .dockerignore
+     * files and therefore will be more efficient
+     * @param relativePathFromBuildRoot
+     */
     public ImageFromDockerfile withDockerfilePath(String relativePathFromBuildRoot) {
```

Comment:
I wonder, would it be possible to make the existing `withDockerfilePath` take advantage of `buildImageCmd::withDockerfile` as well? I think it would be nice to change this PR so that:

* Existing users of the `withDockerfilePath` method get the benefits of `.dockerignore`
* We don't need to add (an effectively duplicate) method for `withDockerfile(String pathname)`

Reply:
Good suggestion, I had thought of this too and experimented a bit initially. Unfortunately I don't think this is feasible because we'd have to take into account things like setting the docker build path. I'm not sure if I'm missing any more, but I'm not familiar enough with the docker-java API (yet) to confidently make this change with certainty it will not break existing users.

Reply:
Shoud we deperecate the old method in this case?

Reply:
That makes sense @aguibert - I can see how complexity could run away in trying to make it work.

I agree with @kiview re deprecating the existing method